### PR TITLE
Add syslog configuration

### PR DESCRIPTION
**Pull request by Filirom1**
Hi,

I wrote the puppet parts corresponding to this blog post <https://blog.openshift.com/central-log-management-openshift-enterprise/> to send OpenShift logs to a central rsyslog server.

The following logs are forwarded:
* openshift-broker
* openshift-console
* broker messages (/var/log/messages)
* node apache
* node application logs (logshifter)
* node application metrics
* openshift node platform
* node messages (/var/log/messages)

BSN (mongodb, activemq) is not supported. Mongodb configuration is quite tricky to change.

I focused on OpenShift Enterprise 2.2, because the package `rsyslog7-mmopenshift` is not available on OpenShift Origin.

Cheers

### CODE REVIEW

**Comment by sdodson** [comment]
Looks good, do you mind putting a link to the blog post in the README.ascii and init.pp? Sorry about the duplication between the two.

**Comment by Filirom1** [comment]
Done ;-)

**Comment by sdodson** [comment]
[test]

**Comment by Filirom1** [comment]
Do you have any idea why it failed ?

**Comment by sdodson** [comment]
Looks like perhaps we're affecting gear logging by default? The first failure I see is this 


{"api_version":1.7,"data":null,"messages":[{"exit_code":157,"field":null,"index":null,"severity":"debug","text":"\n"},{"exit_code":157,"field":null,"index":null,"severity":"warning","text":"Starting MySQL 5.1 cartridge\nMySQL server failed to start:\n"},{"exit_code":157,"field":null,"index":null,"severity":"result","text":"\nMySQL 5.1 database added.  Please make note of these credentials:\n\n       Root User: admindaH4j28\n   Root Password: IZVysiMwN3BU\n   Database Name: app\n\nConnection URL: mysql://$OPENSHIFT_MYSQL_DB_HOST:$OPENSHIFT_MYSQL_DB_PORT/\n\nYou can manage your new MySQL database by also embedding phpmyadmin.\nThe phpmyadmin username and password will be the same as the MySQL credentials above.\n"},{"exit_code":157,"field":null,"index":null,"severity":"error","text":"Failed to execute: 'control start' for /var/lib/openshift/54ae9b473d77ceee89000010/mysql\ncat: /var/lib/openshift/54ae9b473d77ceee89000010/app-root/logs//mysql.log: No such file or directory\n"}],"status":"unprocessable_entity","supported_api_versions":[1.0,1.1,1.2,1.3,1.4,1.5,1.6,1.7],"type":null,"version":"1.7"}

File: manifests/rsyslog.pp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        notify  => Service['rsyslog']
+      }
+    }
+
+    service { 'rsyslog':
+      ensure  => running,
+      enable  => true,
+      require => Exec['install_rsyslog7']
+    }
+  }else{
```
Comment (sdodson) [inline manifests/rsyslog.pp]: I'm not sure we should be disabling rsyslog if $::openshift_origin::syslog_enabled is undef. I noticed in the test run it's stopping rsyslog.

Comment (Filirom1) [inline manifests/rsyslog.pp]: Yes, you're right. I checked on an OpenShift Node without log cetralization, it still needs rsyslog started.

I will remove this statement.

Thanks you !

**Comment by sdodson** [comment]
[test]

**Comment by Filirom1** [comment]
Its worked, thanks ! :beers:

**Comment by sdodson** [comment]
:+1:

**Pull request merged**